### PR TITLE
Implement back button handler, refs #12

### PR DESCRIPTION
**Pull request by lex111**
#12

### CODE REVIEW

File: Navigator.js [hunk 1]
```diff
@@ -85,9 +85,34 @@ export const AppNavigator = StackNavigator(
...
+        BackHandler.addEventListener('hardwareBackPress', () => {
+            const {dispatch, navigation} = this.props;
+
+            if (navigation.index === 0) {
+                return false;
+            }
+                
+            dispatch(NavigationActions.back());
+
+            return true;
```
Comment (ovr) [inline Navigator.js]: Why you are going to return true? Looks likes it's not needed

File: Navigator.js [hunk 2]
```diff
@@ -85,9 +85,34 @@ export const AppNavigator = StackNavigator(
     }
 );
 
-const AppWithNavigationState = ({ dispatch, navigation }) => (
-    <AppNavigator navigation={addNavigationHelpers({ dispatch, state: navigation })} />
-);
+class AppWithNavigationState extends React.Component {
+    componentDidMount() {
+        BackHandler.addEventListener('hardwareBackPress', () => {
+            const {dispatch, navigation} = this.props;
```
Comment (ovr) [inline Navigator.js]: const { a }, space required ;)

File: Navigator.js (hunk 1 above)
Comment (lex111) [inline Navigator.js]: Yes, it is needed, without it will not work

File: Navigator.js (hunk 2 above)
Comment (lex111) [inline Navigator.js]: Exactly, corrected. Can we connect Travis or something like it?

Comment (ovr) [inline Navigator.js]: TravisCI does not check code style now, will be in the feature

Comment (lex111) [inline Navigator.js]: Ok.

**Comment by ovr** [comment]
Thank you @lex111, `cherry-pick`-ed to `master` branch and tested 😸 
Take it 🍺 🍻 🍰

**Pull request closed without merging**